pkg/ebpf/events_pipeline.go

```diff
@@ -240,22 +240,22 @@ func (t *Tracee) sinkEvents(ctx gocontext.Context, in <-chan *trace.Event) <-cha
 	errc := make(chan error, 1)
 
 	go func() {
-		for event := range in {
-			// Only emit events requested by the user
-			if t.events[int32(event.EventID)].emit {
-				if t.config.Output.ParseArguments {
-					err := t.parseArgs(event)
-					if err != nil {
-						t.handleError(err)
-						continue
+		for {
+			select {
+			case event := <-in:
+				// Only emit events requested by the user
+				if t.events[int32(event.EventID)].emit {
+					if t.config.Output.ParseArguments {
+						err := t.parseArgs(event)
+						if err != nil {
+							t.handleError(err)
+							continue
+						}
+						t.config.ChanEvents <- *event
+						t.stats.EventCount.Increment()
+						event = nil
 					}
 				}
-			}
-
-			select {
-			case t.config.ChanEvents <- *event:
-				t.stats.EventCount.Increment()
-				event = nil
 			case <-ctx.Done():
 				return
 			}
```

Comment:
This select statement can be under the `if t.events[int32(event.EventID)].emit {`, which should fix the issue instead of the proposed change